### DISCUSSION TITLE
Layout.vue with multiple slots

### DISCUSSION THREAD

**Discussion by pkpowell** [nuxt/nuxt#18577] [Questions] [answered]
I have a basic three column website with a couple of routes.
The layout has three slots where data from the routes is injected. I have a mockup here: https://stackblitz.com/edit/nuxt-starter-wbcaoc?file=app.vue
The issue is that depending on `app.vue` file I either get smooth logo transitions but no slot content on route changes
```
<template>
  <NuxtLayout>
    <NuxtPage />
  </NuxtLayout>
</template>
```
or, all the expected content but no transitions as the logo seems to be rerendered and flashes on route changes.
```
<template>
    <NuxtPage />
</template>
```
It seems like `<NuxtLayout>` references a generic layout and not the one in `layouts/default.vue` but I think I need to use `<NuxtLayout>` to wrap the multiple <template #> tags.
Am I just doing this wrong?  Is there an other way to wrap templates?

**Comment by ndragun92** [comment, marked as answer]
Hello

I have made a funny example that works but is far far from ideal :) 

https://stackblitz.com/edit/nuxt-starter-vxeij6

**Comment by pkpowell** [comment]
Thanks Nemanja!
It works. Why is it far from ideal? Are there performance issues?

**Reply by ndragun92** [reply]
Well, my example is probably not Server Side friendly regarding its content since I did wrap the layout slot in a client-only tag... but other than that it can be one of the solutions you can approach... I am sure there are better solutions out there as well
